{{cookiecutter.project_slug}}/src/app.js

```diff
@@ -44,6 +51,89 @@ export default class App extends React.Component {
     </Layout>
   );
 
+  //1
+  async checkPermission() {
+    const enabled = await firebase.messaging().hasPermission();
+    if (enabled) {
+      this.getToken();
+    } else {
+      this.requestPermission();
+    }
+  }
+
+  //3
+  async getToken() {
+    let fcmToken = await AsyncStorage.getItem("fcmToken");
```

Comment:
We use redux saga and will use redux-persist for handling AsyncStorage, but for now it is an over kill, so we are ok with this.

But please add a note to indicate that asyncstorage should not be used like this.

Reply:
Got it